thunder/tests/test_core.py

```diff
@@ -3125,3 +3126,32 @@ def test_proxy_same_name():
         t = TensorProxy(name="test", shape=(1,), device=cpu, dtype=float32, requires_grad=True)
         with pytest.raises(RuntimeError, match="already used"):
             t2 = TensorProxy(name="test", shape=(1,), device=cpu, dtype=float32, requires_grad=True)
+
+
+def test_save_trace():
+    def fn(x):
+        return x + 1
+
+    jfn = thunder.jit(fn)
+    jfn(
+        torch.rand(
+            3,
+        )
+    )
+
+    fwd_trace = thunder.last_traces(jfn)[-1]
+
+    trace_filename = "temp_trace.py"
```

Comment:
Please use tempfile: https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile for this. Thank you!